### PR TITLE
Initial PR

### PR DESCRIPTION
**Pull request by svenwltr**
- [x] useful
- [x] tested
- [x] well tested

### CODE REVIEW

**Comment by svenwltr** [comment]
build it

**Comment by svenwltr** [comment]
@rebuy-de/it-platform This works so far. See also rebuy-de/email-silo#42.

**Comment by bjoernhaeuser** [comment]
What do you think about making this not dependent on spring?

**Comment by svenwltr** [comment]
mkay

**Comment by svenwltr** [comment]
@bjoernhaeuser done

File: src/main/java/com/rebuy/consul/RebuyConsulClient.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package com.rebuy.consul;
+
+import com.ecwid.consul.v1.ConsulClient;
+import com.ecwid.consul.v1.agent.model.NewService;
+
+import java.util.ArrayList;
+
+public class RebuyConsulClient
+{
+    public final static String CONSUL_ADDRESS = "192.168.33.10";
```
Comment (bjoernhaeuser) [inline src/main/java/com/rebuy/consul/RebuyConsulClient.java]: these things should be configurable, right?

Comment (svenwltr) [inline src/main/java/com/rebuy/consul/RebuyConsulClient.java]: Hmm ... can we just use `localhost` here, since there is an agent on every host?

Comment (florian-zeidler) [inline src/main/java/com/rebuy/consul/RebuyConsulClient.java]: No, we can't because when the silo is run locally against vagrant it will fail.

Comment (svenwltr) [inline src/main/java/com/rebuy/consul/RebuyConsulClient.java]: I don't get it.

File: src/main/java/com/rebuy/consul/RebuyConsulClient.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        service.setName(name);
+        service.setPort(port);
+        service.setCheck(check);
+        service.setTags(new ArrayList<>());
+        service.getTags().add("consul-java-client");
+
+        ConsulClient client = new ConsulClient(host);
+        client.agentServiceRegister(service);
+
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```
Comment (bjoernhaeuser) [inline src/main/java/com/rebuy/consul/RebuyConsulClient.java]: I would suggest to expose this hook as a seperate method on the client

File: src/main/java/com/rebuy/consul/RebuyConsulClient.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            port
+        ));
+
+        NewService service = new NewService();
+        service.setId(id);
+        service.setName(name);
+        service.setPort(port);
+        service.setCheck(check);
+        service.setTags(new ArrayList<>());
+        service.getTags().add("consul-java-client");
```
Comment (bjoernhaeuser) [inline src/main/java/com/rebuy/consul/RebuyConsulClient.java]: shouldnt that also be configurable? ;)

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+    <name>consul-java-client</name>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <mvn.compiler.version>3.1</mvn.compiler.version>
+        <mvn.source.version>2.2.1</mvn.source.version>
+        <mvn.release.version>2.4.1</mvn.release.version>
+        <spring.version>3.2.3.RELEASE</spring.version>
```
Comment (bjoernhaeuser) [inline pom.xml]: that is unused :)

**Comment by bjoernhaeuser** [comment]
:+1:

File: src/main/java/com/rebuy/consul/RebuyConsulClient.java (hunk 3 above)
Comment (svenwltr) [inline src/main/java/com/rebuy/consul/RebuyConsulClient.java]: This tag is intended to identify that a certain service is using this library. Not sure if this is useful though ...

**Comment by florian-zeidler** [comment]
If we update the consul api to 1.1.8 we do not have the dependency to commons-logging anymore and we do not need jcl-over-slf4j

**Comment by svenwltr** [comment]
So we can also remove the exclusions?

**Comment by florian-zeidler** [comment]
Of course see http://search.maven.org/#artifactdetails|com.ecwid.consul|consul-api|1.1.8|jar

**Comment by svenwltr** [comment]
I am thinking about rewrite this to a builder. Like `ConsulClient.host("foo").registerShutdownHook().tag("blubber").init();`, because testing the current code is gross.

**Comment by florian-zeidler** [comment]
You could do that with the complete config for consul, sounds good

**Comment by svenwltr** [comment]
This Consul API only supports a single check per service :scream: 

https://github.com/Ecwid/consul-api/blob/master/src/main/java/com/ecwid/consul/v1/agent/model/NewService.java#L139

**Comment by florian-zeidler** [comment]
Well, we could open a pull request, if you like?

**Comment by svenwltr** [comment]
I am not sure, if we need this for our library.

**Comment by svenwltr** [comment]
@rebuy-de/it-platform Any thoughts?

File: src/main/java/com/rebuy/consul/ConsulClientBuilder.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+    NewService buildService()
+    {
+        NewService service = new NewService();
+        service.setId(id());
+        service.setName(name());
+        service.setPort(port());
+        service.setCheck(buildCheck());
+        service.setTags(new ArrayList<>());
+        service.getTags().add("consul-java-client");
```
Comment (bjoernhaeuser) [inline src/main/java/com/rebuy/consul/ConsulClientBuilder.java]: shouldnt that be also confiugrable?

**Comment by bjoernhaeuser** [comment]
Looks nice to me

**Pull request merged**